File: App/src/app/shared/entities/teamProject.entity.ts
import { ProjectEntity } from '../../project/entities/project.entity';
import { TeamEntity } from '../../team/entities/team.entity';
import { BaseEntity } from './base.entity';

export interface TeamProjectEntity extends BaseEntity {
  teamId: number;
  team?: TeamEntity;
  projectId: number;
  project?: ProjectEntity;
}

File: Api/src/Api.Domain/Entities/TeamProjectEntity.cs
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace Domain.Entities {
    public class TeamProjectEntity : BaseEntity
    {
        [Required]
        public int TeamId { get; set; }

        [JsonIgnore]
        public TeamEntity Team { get; set; }

        [Required]
        public int ProjectId { get; set; }

        [JsonIgnore]
        public ProjectEntity Project { get; set; }
    }
}
